test/compress/evaluate.js

```diff
@@ -37,3 +37,351 @@ positive_zero: {
+        console.log(
+            ({0:1}) + 1,
+            ({0:1})[0] + 1,
+            ({0:1})["0"] + 1,
+            ({0:1})[1] + 1,
+            ({0:1})[0][1] + 1,
+            ({0:1})[0]["1"] + 1
+        );
+    }
+    expect: {
+        console.log(
+            ({0:1}) + 1,
+            2,
+            2,
+            ({0:1})[1] + 1,
+            1[1] + 1,
+            1["1"] + 1
+        );
+    }
+}
+
+unsafe_integer_key_complex: {
+    options = {
+        evaluate  : true,
+        unsafe    : true
+    }
+    input: {
+        console.log(
+            ({0:{1:1},1:1}) + 1,
+            ({0:{1:1},1:1})[0] + 1,
+            ({0:{1:1},1:1})["0"] + 1,
+            ({0:{1:1},1:1})[1] + 1,
+            ({0:{1:1},1:1})[0][1] + 1,
+            ({0:{1:1},1:1})[0]["1"] + 1
+        );
+    }
+    expect: {
+        console.log(
+            ({0:{1:1},1:1}) + 1,
+            "[object Object]1",
+            "[object Object]1",
+            2,
+            2,
+            2
+        );
+    }
+}
+
+unsafe_float_key: {
+    options = {
+        evaluate  : true,
+        unsafe    : true
+    }
+    input: {
+        console.log(
+            ({2.72:1}) + 1,
+            ({2.72:1})[2.72] + 1,
+            ({2.72:1})["2.72"] + 1,
+            ({2.72:1})[3.14] + 1,
+            ({2.72:1})[2.72][3.14] + 1,
+            ({2.72:1})[2.72]["3.14"] + 1
+        );
+    }
+    expect: {
+        console.log(
+            ({2.72:1}) + 1,
+            2,
+            2,
+            ({2.72:1})[3.14] + 1,
+            1[3.14] + 1,
+            1["3.14"] + 1
+        );
+    }
+}
+
+unsafe_float_key_complex: {
+    options = {
+        evaluate  : true,
+        unsafe    : true
+    }
+    input: {
+        console.log(
+            ({2.72:{3.14:1},3.14:1}) + 1,
+            ({2.72:{3.14:1},3.14:1})[2.72] + 1,
+            ({2.72:{3.14:1},3.14:1})["2.72"] + 1,
+            ({2.72:{3.14:1},3.14:1})[3.14] + 1,
+            ({2.72:{3.14:1},3.14:1})[2.72][3.14] + 1,
+            ({2.72:{3.14:1},3.14:1})[2.72]["3.14"] + 1
+        );
+    }
+    expect: {
+        console.log(
+            "[object Object]1",
+            "[object Object]1",
+            "[object Object]1",
+            2,
+            2,
+            2
+        );
+    }
+}
+
+unsafe_array: {
+    options = {
+        evaluate  : true,
+        unsafe    : true
+    }
+    input: {
+        console.log(
+            [1, 2, 3, a] + 1,
+            [1, 2, 3, 4] + 1,
+            [1, 2, 3, a][0] + 1,
+            [1, 2, 3, 4][0] + 1,
+            [1, 2, 3, 4][6 - 5] + 1,
+            [1, , 3, 4][6 - 5] + 1,
+            [[1, 2], [3, 4]][0] + 1,
+            [[1, 2], [3, 4]][6 - 5][1] + 1,
+            [[1, 2], , [3, 4]][6 - 5][1] + 1
+        );
+    }
+    expect: {
+        console.log(
+            [1, 2, 3, a] + 1,
+            "1,2,3,41",
+            [1, 2, 3, a][0] + 1,
+            2,
+            3,
+            NaN,
+            "1,21",
+            5,
+            (void 0)[1] + 1
```

Comment:
Could you please add a additional simpler test for an array hole that just results in `void 0`? 
Something like: `[10,,30][1]`